kubernetes/structures.go

```diff
@@ -26,6 +26,15 @@ func buildId(meta metav1.ObjectMeta) string {
 	return meta.Namespace + "/" + meta.Name
 }
 
+func buildIdWithVersionKind(meta metav1.ObjectMeta, kind, apiVersion string) string {
+	id := fmt.Sprintf("apiVersion=%v,kind=%v,name=%s",
+		apiVersion, kind, meta.Name)
+	if meta.Namespace != "" {
+		id += fmt.Sprintf(",namespace=%v", meta.Namespace)
+	}
+	return id
+}
+
```

Comment:
We have a function for parsing this in the manifest provider, but I'll need to shuffle some things around to share it between both providers.